packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -644,18 +644,19 @@ class DragTarget<T extends Object> extends StatefulWidget {
   _DragTargetState<T> createState() => _DragTargetState<T>();
 }
 
-List<T?> _mapAvatarsToData<T extends Object>(List<_DragAvatar<T>> avatars) {
-  return avatars.map<T?>((_DragAvatar<T> avatar) => avatar.data).toList();
+List<T?> _mapAvatarsToData<T extends Object>(List<_DragAvatar<Object>> avatars) {
+  return avatars.map<T?>((_DragAvatar<Object> avatar) => avatar.data as T?).toList();
 }
 
 class _DragTargetState<T extends Object> extends State<DragTarget<T>> {
-  final List<_DragAvatar<T>> _candidateAvatars = <_DragAvatar<T>>[];
+  final List<_DragAvatar<Object>> _candidateAvatars = <_DragAvatar<Object>>[];
   final List<_DragAvatar<Object>> _rejectedAvatars = <_DragAvatar<Object>>[];
 
   bool didEnter(_DragAvatar<Object> avatar) {
     assert(!_candidateAvatars.contains(avatar));
     assert(!_rejectedAvatars.contains(avatar));
-    if (avatar is _DragAvatar<T> && (widget.onWillAccept == null || widget.onWillAccept!(avatar.data))) {
+    if ((avatar.data == null || avatar.data is T)
+        && (widget.onWillAccept == null || widget.onWillAccept!(avatar.data as T?))) {
```

Comment:
This fixes:
DragTarget<int>` should accept `Draggable<Object>` data(runtime is `int`)

Before this change,  `avatar` runtime is `_DragAvatar<Object>` which is not a subtype of `_DragAvatar<int>`, cause not be accepted